static/sass/_snapcraft_l-application.scss

```diff
@@ -127,64 +107,85 @@
     }
 
     .store-selector__panel {
-      background-color: #2d2d2d;
+      background-color: var(--vf-color-background-hover);
+      border: 1px solid var(--vf-color-border-default);
       left: 0;
-      padding: 0.5rem 0.5rem 0;
       position: absolute;
       top: 2.3rem;
       width: 100%;
       z-index: 1;
     }
 
     .store-selector__list {
-      background-color: #2d2d2d;
       margin: 0;
       padding-left: 0;
     }
 
     .store-selector__item {
-      border-bottom: 1px solid rgba(255, 255, 255, 0.25);
-      padding: calc(0.4rem - 1px) 0;
+      border-bottom: 1px solid var(--vf-color-border-muted);
 
       &:last-child {
         border-bottom: 0;
       }
 
       a {
-        color: $color-x-light;
+        color: var(--vf-color-text-default);
         display: block;
         overflow: hidden;
+        padding: calc(0.4rem - 1px) 0.5rem;
         text-overflow: ellipsis;
         white-space: nowrap;
 
         &:hover {
+          background-color: var(--vf-color-background-default);
           text-decoration: none;
         }
       }
     }
   }
 
-  // Add collapsible nav in application layout
-  @include app-p-icon;
+  .l-navigation {
+    &.is-collapsed {
```

Comment:
What's the reason for those overrides we are introducing? If we need to have them, they should be clearly commented in the code to understand what they are needed for.
With any override like this we risk that the upstream updates will not work as expected, so it's important to understand why each override is needed.

Maybe overrides can be avoided if we add some custom variant class name that is specific to our case? This could potentially also avoid using `!important` that is generally a bad practice and something that prevents CSS to work in native way.